### ISSUE TITLE
[FEATURE] Add option `load_best` to `TorchForecastingModel.fit()` and `fit_from_dataset()`

### ISSUE THREAD

**Issue by dennisbader** [unit8co/darts#2902]
Related to #2064.

### Summary

We should add an option to our TorchForecastinModel that allows to automatically load (and use) the best model on the validation set at the end of the training process. 

This should only be effective if `val_series` and `val_*_covaraites` are provided, and `save_checkpoints=True` at model creation.

### Potential solution
We could add a param `load_best` (?) to `fit()` and `fit_from_dataset()` to control this.

### Alternative solution
We could instead also add a new param to `__init__()` but I feel it's more appropriate in the `fit*()` methods. Happy to discuss.

**Comment by cnhwl** [comment]
Hello! @dennisbader I'd be delighted to complete this feature. PR has been made and happy to hear your code review.

**Comment by dennisbader** [comment]
Thanks a lot for all the contributions @cnhwl it's really appreciated :) 🚀
